### ISSUE TITLE
Portability?

### ISSUE THREAD

**Issue by Technerder** [MycroftAI/mycroft-precise#71]
How portable is this project, and which binaries would I need to ship alongside my application so I can avoid having to rebuild the project?

**Comment by MatthewScholefield** [comment]
The [binaries](https://github.com/MycroftAI/mycroft-precise/releases) are self contained using [pyinstaller](https://www.pyinstaller.org/) meaining they only require that the version of GLIBC the user is running is newer than the OS it was built on. Beyond that, there are no other dependencies as they are all packaged inside the executable.

As for what you would need to ship with your application, you would need to include the `precise-engine_VERSION_x86_64.tar.gz` data somehow so that you could execute the `precise-engine/precise-engine` file inside. Typically, this is easiest done using [our Python wrapper](https://github.com/MycroftAI/mycroft-precise/tree/dev/runner) which is a python module that only uses pyaudio.

Let me know if this answers what you were asking.o

**Comment by Technerder** [comment]
So assuming that the system I'm wanting this to run on has GLIBC installed, I would only have to have the ``precise-engine`` file I built during the source install for it to work?

**Comment by MatthewScholefield** [comment]
No, you would only have to have the extracted precise-engine_0.3.0_x86_64.tar.gz file that's in the releases tab of this repo. Alternatively, you could build this file yourself using the command ./build.sh (and it would be placed in the dist/ folder)
